include/seqan3/io/all.hpp

```diff
@@ -12,45 +12,90 @@
  *
- * <small>¹ SeqAn always assumes GZip and does not handle pure `.Z`.<br>
- * ²Some file formats like `.bam` or `.bcf` are implicitly BGZF-compressed without showing this in the
- * extension.</small>
+ * For example, seqan3::sequence_file_input handles reading sequence files.
+ * It can be created directly from an input stream, or from a file name. After opening the file it will detect whether
+ * the format is seqan3::format_fasta or seqan3::format_fastq (or another supported format) automatically – normally
+ * by comparing the extension.
  *
- * Whether or not a format is supported depends on whether the respective dependency was available and
- * linked against your program (if you use CMake, this should happen automatically).
+ * | **File**                      | **Formats**                                                                                                  |
+ * |:------------------------------|:-------------------------------------------------------------------------------------------------------------|
+ * | seqan3::alignment_file_input  | seqan3::format_sam, seqan3::format_bam                                                                       |
+ * | seqan3::alignment_file_output | seqan3::format_sam, seqan3::format_bam                                                                       |
+ * | seqan3::sequence_file_input   | seqan3::format_embl, seqan3::format_fasta, seqan3::format_fastq, seqan3::format_genbank, seqan3::format_sam  |
+ * | seqan3::sequence_file_output  | seqan3::format_embl, seqan3::format_fasta, seqan3::format_fastq, seqan3::format_genbank, seqan3::format_sam  |
+ * | seqan3::structure_file_in  | seqan3::format_vienna                                                                                        |
+ * | seqan3::structure_file_out | seqan3::format_vienna                                                                                        |
  *
- * Formatted files employ compression/decompression streams transparently, i.e. if the given file-extension or
- * "magic-header" of a file suggest this, the respective stream is automatically (de-)compressed.
+ * Some formats are available in multiple files, e.g. seqan3::format_sam can be read by seqan3::sequence_file_input
+ * and by seqan3::alignment_file_input. This represents different use-cases of the same file format.
+ *
+ * Typically formats are supported for reading and writing, but this does not always have to be the case. See the above
+ * links for more information.
+ *
+ * ## Records and fields
+ *
+ * The main file interface that SeqAn offers is *record-based*, i.e. every file conceptionally is a range of
+ * records. And each record in turn behaves as a tuple of fields.
+ *
+ * The record type of all files is based on seqan3::record, but the composition of fields is different **per file**.
+ *
+ * Maybe a little more concrete:
```

Comment:
```suggestion
 * In particular this means:
```
Or something similar, it sounds quite colloquial as it is.

Reply:
Ok. I have another documentation PR in the pipeline and will update this inside their so we don't have to wait so much for travis, ok?